core/criteria/src/main/java/org/opennms/core/criteria/CriteriaBuilder.java

```diff
@@ -86,15 +91,36 @@ public Criteria toCriteria() {
         criteria.setDistinct(m_distinct);
         criteria.setLimit(m_limit);
         criteria.setOffset(m_offset);
-
+        criteria.setMultipleAnd(m_isMultipleAnd);
         if ("any".equals(m_matchType)) {
             criteria.setRestrictions(Collections.singleton(Restrictions.any(m_restrictions.toArray(EMPTY_RESTRICTION_ARRAY))));
         } else {
             criteria.setRestrictions(m_restrictions);
         }
-
+        if(isNestedMultipleAnd(m_restrictions)){
+            throw new IllegalArgumentException("Use of nested 'multiAnd' is not allowed");
+        }
         return criteria;
     }
+    private boolean isNestedMultipleAnd(Collection<? extends Restriction> allRestrictions){
+        AtomicBoolean result = new AtomicBoolean(false);
+        //set of multiand allRestrictions
+        Set<Restriction> multiAndRestrictionSet = allRestrictions.stream().filter(
+                restriction -> restriction.getType().equals(Restriction.RestrictionType.MULTIAND)).collect(Collectors.toSet());
+        //iterate over "multiAnd"
+        multiAndRestrictionSet.stream().forEach(restriction ->{
+            //Get all inner restrictions
+            Collection<Restriction> allMultiAndRestrictions = ((AllRestriction) restriction).getRestrictions();
+            //iterate over inner restrictions
+            allMultiAndRestrictions.stream().forEach(singleMultiAndRestriction ->{
+                //check if any "multiAnd" present, then set result true
+                if(singleMultiAndRestriction.getType().equals(Restriction.RestrictionType.MULTIAND)){
+                    result.set(true);
+                }
+            });
+        });
+        return result.get();
```

Comment:
May be this can be simplified with
```suggestion
                Set<Restriction> allInnerRestrictions = multiAndRestrictionSet.stream().flatMap(restriction -> ((AllRestriction) restriction).getRestrictions().stream()).collect(Collectors.toSet());
        return allInnerRestrictions.stream().anyMatch(restriction -> restriction.getType().equals(Restriction.RestrictionType.MULTIAND));
```

Good to add a test case for this

Reply:
Optimized code added

Reply:
Test case (testGetEventsForEventParameters_Case4) is already there to test nestedAnd condition